pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperDataCache.java

```diff
@@ -143,7 +143,7 @@ public void process(WatchedEvent event) {
         }
     }
 
-    public void close() {
+    public synchronized void close() {
```

Comment:
I just removed the synchronized, since it is not clear whether adding it to process() is lightweight. We can live with the small risk of a broker needing to be restarted if and when the load manager implementation is changed.